### ISSUE TITLE
Poor performance of BlackJax SgLD for Bayesian CNN on MNIST dataset.

### ISSUE THREAD

**Issue by HitarthGandhi** [blackjax-devs/blackjax#516]
We are trying to use BlackJAX SgLD for BCNN on the MNIST dataset to solve blackjax-devs/sampling-book#14. We used the [SgMCMCJax](https://sgmcmcjax.readthedocs.io/en/latest/nbs/Flax_MNIST.html) library for SgLD, which gives an accuracy of 94.4%. BlackJax is unable to reproduce the results in the same setting. The detailed code and our findings can be found in [this notebook](https://colab.research.google.com/drive/1T4MS92qP7o0v3ru4jY-TUSEdtzFrQIiO#scrollTo=9sJs5GKc4Lyj).

You can see in the below plot that BlackJax SGLD does not give the desired results.
<img src= https://user-images.githubusercontent.com/61594952/231424270-b8fdcc18-f16e-4e3c-bc6a-15d72720de65.png width=600/>

**Comment by albcab** [comment]
Try installing the latest developments with `pip install blackjax-nightly`. Notice that you'll need to change your code a bit for it to work with the latest SgLD kernel. Specifically, you won't be initializing a state and instead passing the previous position directly to the step function, you'll also be passing the `learning_rate`/`step_size` to the step function instead of the kernel initializer:
```python
sgld = blackjax.sgld(grad_fn)
new_position = sgld.step(rng_key, position, minibatch, step_size)
```
Hope this helps, let us know if it works. We're working on a new release soon.
